src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -726,7 +726,26 @@ internal override void Complete()
 
                     this.Command.Context.SetVariable(SpecialVariables.LastExitCodeVarPath, _nativeProcess.ExitCode);
                     if (_nativeProcess.ExitCode != 0)
+                    {
                         this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
```

Comment:
If we throw a terminating error, do we need to execute this line?  If not, I could move this line after the test if we should throw.

Reply:
This depends on NativeCommandThrowPreference. If PSNativeCommandUseErrorActionPreference  we keep old behavior.

Reply:
I think my question does not apply anymore since the change I made.  This code will now only `Write` an ErrorRecord and not call ThrowTerminatingError().